### DISCUSSION TITLE
Branch/sublist/subvalues accessor/iterator

### DISCUSSION THREAD

**Discussion by ghost** [SkriptLang/Skript#4028] [Suggestions]
Here is a proposed syntax:
``{some_list::*::same_field_available_in_all_elements}``
It returns a list of the field requested from all the elements of the list. In case a field doesn't exist in the branch it adds "none" as to not loose the order/total amount of elements. Example of usability:

Data structure:
```
clanData | -> clan1 | -> name: "MegaMonsters"
                    | -> memberCount: 4
                             
         | -> clan2 | -> name: "ObsidianMasters"
                    | -> memberCount: 3
```
Get all the names of the clans:
```set {_names::*} to {clanData::*::name}```

I talked with Snow-Pyon on discord and they suggested it might not be clear that said syntax returns a list and gave some alternatives:

1.
```
loop tree of {some-var::*}:
    send {some-var::%branch%}
```

2.
```
                           |--> attack_speed = 1.0
attributes |--> Snow-Pyon --
           |               |--> max_health = 20.0
           |
           |               |--> attack_speed = 2.0
           |-->  Runakai  --
                           |--> max_health = 40.0
```
```set {_values::*} to branch "attack_speed" of {attributes::*}```

**Comment by Mwexim** [comment]
I find the very first syntax (being `{foo::*::bar}`) very descriptive: it immediately tells you what is going on. While this may be confusing to new users, this is absolutely the easiest way to implement it and should, in my opinion, at least be an **option**.
